cpp/arcticdb/stream/segment_aggregator.hpp

```diff
@@ -43,9 +43,13 @@ template<class Index, class Schema, class SegmentingPolicy = RowCountSegmentPoli
         SliceCallBack&& slice_callback,
         Schema &&schema,
         typename AggregatorType::Callback &&c,
-        SegmentingPolicy &&segmenting_policy = SegmentingPolicy{}) :
```

Comment:
I think that the segment aggregator is already quite overloaded in terms of the number of use-cases it supports, and the collector-restart dedup stuff should not go back into it. It would be better to have a collector-specific aggregator in the other project, even if there is some code duplication at first, as I think they are going to diverge quite significantly and it's going to be really hard if all the changes have one foot in each repo